src/sentry/search/events/datasets/metrics.py

```diff
@@ -2018,3 +2045,28 @@ def _resolve_rate(
             ],
             alias,
         )
+
+    def _resolve_http_response_count(
+        self,
+        args: Mapping[str, str | Column | SelectType | int | float],
+        alias: str | None = None,
+    ) -> SelectType:
+        condition = Function(
+            "startsWith",
+            [
+                self.builder.column("span.status_code"),
+                args["code"],
+            ],
+        )
+
+        return self._resolve_count_if(
+            Function(
+                "equals",
+                [
+                    Column("metric_id"),
+                    self.resolve_metric("span.self_time"),
+                ],
+            ),
+            condition,
+            alias,
+        )
```

Comment:
This looks quite identical to https://github.com/getsentry/sentry/blob/3860aba65a6d22152316f304bb2e13de18036eda/src/sentry/search/events/datasets/spans_metrics.py#L895-L918

Does it make sense to just share this function between metrics and spans_metrics?

Reply:
Good point. I took a look at consolidating this and ran into some issues since this function uses other functions that are further duplicated in both datasets. I think I might save this for another PR since the code changes involved are more than expected